### ISSUE TITLE
Crash on clicking home button in contextual menu

### ISSUE THREAD

**Issue by VishalNehra** [TeamAmaze/AmazeFileManager#2138]
**Smartphone:**
- Device: ASUS_X00P_4
- OS: Linux Android 10 - 29
- Rooted: Maybe
- Version: 3.5.1
### Stack traces:
**Trace:**
```java
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:399)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.IllegalStateException: Unexpected failure to classify SearchResult: SearchResult{key=/storage/emulated/0/.UTSystemConfig```
**Trace:**
```java
 nodeFound=Node{edge=/storage/emulated/0/.ADM```
**Trace:**
```java
 value=-```
**Trace:**
```java
 edges=[]}```
**Trace:**
```java
 charsMatched=21```
**Trace:**
```java
 charsMatchedInNodeFound=21```
**Trace:**
```java
 parentNode=Node{edge=```
**Trace:**
```java
 value=null```
**Trace:**
```java
 edges=[Node{edge=/storage/emulated/0/.ADM```
**Trace:**
```java
 value=-```
**Trace:**
```java
 edges=[]}]}```
**Trace:**
```java
 parentNodesParent=null```
**Trace:**
```java
 classification=null}
	at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.classify(ConcurrentRadixTree.java:989)
	at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.<init>(ConcurrentRadixTree.java:969)
	at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.searchTree(ConcurrentRadixTree.java:932)
	at com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.getValueForExactKey(ConcurrentRadixTree.java:102)
	at com.amaze.filemanager.utils.DataUtils.isFileHidden(DataUtils.java:359)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.createListParcelables(LoadFilesListTask.java:273)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4(LoadFilesListTask.java:231)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4$LoadFilesListTask(Unknown Source:0)
	at com.amaze.filemanager.asynchronous.asynctasks.-$$Lambda$LoadFilesListTask$UXzmWslcH0cfZfAA6Y9hqkVwFiA.invoke(Unknown Source:6)
	at com.amaze.filemanager.filesystem.root.ListFilesCommand.getFilesList(ListFilesCommand.kt:160)
	at com.amaze.filemanager.filesystem.root.ListFilesCommand.listFiles(ListFilesCommand.kt:64)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:222)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:67)
	at android.os.AsyncTask$3.call(AsyncTask.java:378)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
```
<details><summary><b>Raw mail</b></summary><p>
```json
{
  "request": "Application crash",
  "user_action": "UI Error",
  "product": "WW_X00PD",
  "package": "com.amaze.filemanager",
  "os": "Linux Android 10 - 29",
  "model": "ASUS_X00PD",
  "time": "2020-12-13 18:23",
  "version": "3.5.1",
  "device": "ASUS_X00P_4",
  "exceptions": "[java.lang.RuntimeException: An error occurred while executing doInBackground()\n\tat android.os.AsyncTask$4.done(AsyncTask.java:399)\n\tat java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)\n\tat java.util.concurrent.FutureTask.setException(FutureTask.java:252)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:271)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\n\tat java.lang.Thread.run(Thread.java:919)\nCaused by: java.lang.IllegalStateException: Unexpected failure to classify SearchResult: SearchResult{key=/storage/emulated/0/.UTSystemConfig, nodeFound=Node{edge=/storage/emulated/0/.ADM, value=-, edges=[]}, charsMatched=21, charsMatchedInNodeFound=21, parentNode=Node{edge=, value=null, edges=[Node{edge=/storage/emulated/0/.ADM, value=-, edges=[]}]}, parentNodesParent=null, classification=null}\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.classify(ConcurrentRadixTree.java:989)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree$SearchResult.<init>(ConcurrentRadixTree.java:969)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.searchTree(ConcurrentRadixTree.java:932)\n\tat com.googlecode.concurrenttrees.radix.ConcurrentRadixTree.getValueForExactKey(ConcurrentRadixTree.java:102)\n\tat com.amaze.filemanager.utils.DataUtils.isFileHidden(DataUtils.java:359)\n\tat com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.createListParcelables(LoadFilesListTask.java:273)\n\tat com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4(LoadFilesListTask.java:231)\n\tat com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.lambda$doInBackground$4$LoadFilesListTask(Unknown Source:0)\n\tat com.amaze.filemanager.asynchronous.asynctasks.-$$Lambda$LoadFilesListTask$UXzmWslcH0cfZfAA6Y9hqkVwFiA.invoke(Unknown Source:6)\n\tat com.amaze.filemanager.filesystem.root.ListFilesCommand.getFilesList(ListFilesCommand.kt:160)\n\tat com.amaze.filemanager.filesystem.root.ListFilesCommand.listFiles(ListFilesCommand.kt:64)\n\tat com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:222)\n\tat com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:67)\n\tat android.os.AsyncTask$3.call(AsyncTask.java:378)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\t... 3 more\n]",
  "user_comment": "crashed soon after I clicked on the home button shortcut"
}
```

**Comment by TranceLove** [comment]
Should be duplicate of #2106, which have been fixed by #2192?

**Comment by VishalNehra** [comment]
Yes
